### ISSUE TITLE
KeyError: 'BatchMatMulV2'

### ISSUE THREAD

**Issue by YC-wind** [tensorflow/tensorflow#31837]
KeyError: 'BatchMatMulV2'

train and save model in tf 1.14 environment,
but get  --- KeyError: 'BatchMatMulV2'  ---- in tf 1.13 environment,
how to solve this problem in tf 1.13 environment

**Comment by YC-wind** [comment]
wait for help

**Comment by gadagashwini** [comment]
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by YC-wind** [comment]
when i use below method , load the saved model,  it throw  KeyError: 'BatchMatMulV2' ---- in tf 1.13 environment,  (proble is here: saver = tf.train.import_meta_graph(input_checkpoint + '.meta', clear_devices=clear_devices))

def load_model(model_folder):
    # We retrieve our checkpoint fullpath
    try:
        checkpoint = tf.train.get_checkpoint_state(model_folder)
        input_checkpoint = checkpoint.model_checkpoint_path
        print("[INFO] input_checkpoint:", input_checkpoint)
    except Exception as e:
        input_checkpoint = model_folder
        print("[INFO] Model folder", model_folder, repr(e))

    # We clear devices to allow TensorFlow to control on which device it will load operations
    clear_devices = True
    tf.reset_default_graph()
    # We import the meta graph and retrieve a Saver
    saver = tf.train.import_meta_graph(input_checkpoint + '.meta', clear_devices=clear_devices)

    # We retrieve the protobuf graph definition

    graph = tf.get_default_graph()
    input_graph_def = graph.as_graph_def()

    # We start a session and restore the graph weights
    sess = tf.Session()
    saver.restore(sess, input_checkpoint)

    return sess


but it work in this method ( i think problem is loading the graph  )

def get_assignment_map_from_checkpoint(tvars, init_checkpoint):
    """Compute the union of the current variables and checkpoint variables."""
    assignment_map = {}
    initialized_variable_names = {}

    name_to_variable = collections.OrderedDict()
    for var in tvars:
        name = var.name
        m = re.match("^(.*):\\d+$", name)
        if m is not None:
            name = m.group(1)
        name_to_variable[name] = var

    init_vars = tf.train.list_variables(init_checkpoint)

    assignment_map = collections.OrderedDict()
    for x in init_vars:
        (name, var) = (x[0], x[1])
        if name not in name_to_variable:
            continue
        assignment_map[name] = name
        initialized_variable_names[name] = 1
        initialized_variable_names[name + ":0"] = 1

    return (assignment_map, initialized_variable_names)



finally , i solve the problem by init graph which designed before, 

i also want how to solve the problem ,because  init graph is not elegant enough

**Comment by YC-wind** [comment]
addition:

it  works well in tf 1.14 environment( train and save model in tf 1.14,  can get the graph from meta file )
in tf 1.13 environment, it can not get the graph from meta file, only init graph works.

thanks

**Comment by gadagashwini** [comment]
@YC-wind, can it be possible to share the complete code to reproduce the reported issue. Thanks!

**Comment by gadagashwini** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by taiftahmid** [comment]
I am facing the same issue while trying to convert .meta to pb file in tensorflow version 1.13.1. 

This is the error:
File "/home/taif/.local/lib/python3.7/site-packages/tensorflow/python/framework/importer.py", line 159, in _RemoveDefaultAttrs
    op_def = op_dict[node.op]

KeyError: 'BatchMatMulV2'

The code:
```
import tensorflow as tf

meta_path = 'fns.ckpt.meta' # Your .meta file
output_node_names = ['add_37']    # Output nodes

with tf.Session() as sess:

    # Restore the graph
    saver = tf.train.import_meta_graph(meta_path)

    # Load weights
    saver.restore(sess,"checkpoints/fns.ckpt")

    # Freeze the graph
    frozen_graph_def = tf.graph_util.convert_variables_to_constants(
        sess,
        sess.graph_def,
        output_node_names)

    # Save the frozen graph
    with open('output_graph.pb', 'wb') as f:
      f.write(frozen_graph_def.SerializeToString())
```

**Comment by gadagashwini** [comment]
@taiftahmid,
Can you please post a new issue by providing the information asked by the [template](https://github.com/tensorflow/tensorflow/issues/new/choose)?
The reason for this is we can focus on your specific configuration and problem since the root cause can be unrelated even though the error messages are similar. Thanks!

**Comment by Ningshiqi** [comment]
I came across the same problem with tf 1.13. 
It solved by tf 1.14.
